shub/image/build.py

```diff
@@ -49,19 +53,71 @@ def build_cmd(target, version, skip_tests):
     if not os.path.exists(os.path.join(project_dir, 'Dockerfile')):
         raise shub_exceptions.BadParameterException(
             "Dockerfile is not found, please use shub image 'init' command")
-    is_built = False
-    for data in client.build(path=project_dir, tag=image_name, decode=True):
-        if 'stream' in data:
-            utils.debug_log("{}".format(data['stream'][:-1]))
-            is_built = re.search(
-                r'Successfully built ([0-9a-f]+)', data['stream'])
-        elif 'error' in data:
-            click.echo("Error {}:\n{}".format(
-                data['error'], data['errorDetail']))
-    if not is_built:
-        raise shub_exceptions.RemoteErrorException(
-            "Build image operation failed")
+    if utils.is_verbose():
+        build_progress_cls = _LoggedBuildProgress
+    else:
+        build_progress_cls = _BuildProgress
+    events = client.build(path=project_dir, tag=image_name, decode=True)
+    build_progress = build_progress_cls(events)
+    build_progress.show()
     click.echo("The image {} build is completed.".format(image_name))
     # Test the image content after building it
     if not skip_tests:
         test_cmd(target, version)
+
+
+class _LoggedBuildProgress(utils.BaseProgress):
+    """Visualize build progress in verbose mode.
+
+    Output all the events received from the docker daemon.
+    """
+    def handle_event(self, event):
+        if 'stream' in event:
+            self.handle_stream_event(event)
+        if 'error' in event:
+            tqdm.write("Error {}: {}".format(event['error'],
+                                             event['errorDetail']))
+            raise shub_exceptions.RemoteErrorException(
+                "Build image operation failed")
```

Comment:
I recall while I tested this workflow in the past, I've seen some error events going one after another describing same error with more details, so I followed the stream model - process all the messages until there's something serious that stops yielding it. However I can't reproduce it now, and not sure if it's still applicable.

It was [discussed](https://github.com/docker/docker-py/issues/449) in `docker-py` tracker - and most users just check for `error`/`errorDetail` fields in event - that's what docker-compose [does](https://github.com/docker/compose/blob/master/compose/progress_stream.py#L56) and that's what we do when building an image on deploy.

After this investigation, I think it's fine to consider the change above as a deliberate one 😄  Objections?